### ISSUE TITLE
Improved Description on Transaction Confirmation Chrome Pop-Up

### ISSUE THREAD

**Issue by evabeylin** [MetaMask/metamask-extension#5211]
Suggested improvement on the chrome pop-up confirming a transaction:

Currently the pop-up states "Transaction #X" is confirmed. With X being the number of the transaction from that user's ethereum address. Even better if instead of the transaction number, other metadata could be used as the description for the transaction confirmation - such as the dapp or browser the transaction was occuring on, the token that was transferred etc.

E.g., instead of Alice conducting a 6th transaction from her Ethereum address to purchase a Cryptokitty and the pop-up stating "Transaction 6 Confirmed" --> "Cryptokitties Transaction Confirmed".

Can enable better UX and ensure that if users are conducting ethereum transactions across multiple platforms in the same time frame, they can know which or where the transaction took place.

**Comment by bdresser** [comment]
@evabeylin good idea! The original tx notification was done by a community member and we were just stoked to have something better than the status quo!

It'd be nice to show (1) the name of the contract the user is interacting with and (2) the type of interaction ("sent ether", "approve", etc). It's possible we know neither (i.e. "unknown function"), in which case we should could fallback to the nonce. 

@danfinlay thoughts?

**Comment by danfinlay** [comment]
This design space can be pretty wide open for us. Our `txMeta` class has a lot of potentially useful properties (`txMeta.txParams` are the standard ethereum tx parameters). We could make this say whatever we want right here:

https://github.com/MetaMask/metamask-extension/blob/e2be22a4b722df608cb764042cc8ade6664414d8/app/scripts/platforms/extension.js#L68

I like the idea of incorporating the source site. I also wouldn't mind including a method-name comprehension, maybe using [eth-method-registry](https://www.npmjs.com/package/eth-method-registry).

**Comment by danfinlay** [comment]
It strikes me that this string of text is not using our internationalization. That's bad. Also it probably shouldn't be in the platform definition file, since the wording probably isn't platform-specific. Anyways, just adding notes here.

**Comment by mrwillis** [comment]
@danfinlay How many confirmations does MetaMask wait for to show this chrome extension?
